### ISSUE TITLE
Does not work with FastAPI/starlette's TestClient

### ISSUE THREAD

**Issue by nikolarankovic-tomtom** [miketheman/pytest-socket#47]
Hi, thank you for this excellent plugin. While it works fine with Flask's test client, the plugin blocks starlette's [TestClient](https://www.starlette.io/) (used by [FastAPI](https://fastapi.tiangolo.com/)). Here's a self-contained example that I'd expect to pass with `pytest`:

```python
from pytest_socket import disable_socket
from starlette.responses import HTMLResponse
from starlette.testclient import TestClient


def pytest_runtest_setup():
    disable_socket()


async def app(scope, receive, send):
    assert scope['type'] == 'http'
    response = HTMLResponse('<html><body>Hello, world!</body></html>')
    await response(scope, receive, send)


def test_app():
    client = TestClient(app)
    response = client.get('/')
    assert response.status_code == 200
```

The above returns a:
```
E       pytest_socket.SocketBlockedError: A test tried to use socket.socket
```

Would it be possible to make this work with FastAPI/Starlette?

**Comment by miketheman** [comment]
Hi @nikolarankovic-tomtom ! Thanks for providing the example. I think this might have to do with the `asyncio` usage, similar to what's been reporting in #6 . I haven't gotten around to working on a fix yet.

**Comment by miketheman** [comment]
Closing, as this should be addressed by the changes supplied in #54 
Please reopen with further details if it doesn't solve the issue!
